keyboards/massdrop/alt/keymaps/default/keymap.c

```diff
@@ -151,50 +151,86 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
             return false;
         case U_T_AUTO:
             if (record->event.pressed && MODS_SHIFT && MODS_CTRL) {
-                usb_extra_manual = !usb_extra_manual;
-                CDC_print("USB extra port manual mode ");
-                CDC_print(usb_extra_manual ? "enabled" : "disabled");
-                CDC_print("\r\n");
+                if (usb_extra_manual)
+                {
+                    dprintf("USB extra port manual mode disabled\r\n");
+                    usb_extra_manual = 0;
+                }
+                else
+                {
+                    usb_extra_manual = 1;
+                    dprintf("USB extra port manual mode enabled\r\n");
+                }
             }
             return false;
         case U_T_AGCR:
             if (record->event.pressed && MODS_SHIFT && MODS_CTRL) {
-                usb_gcr_auto = !usb_gcr_auto;
-                CDC_print("USB GCR auto mode ");
-                CDC_print(usb_gcr_auto ? "enabled" : "disabled");
-                CDC_print("\r\n");
+                if (usb_gcr_auto)
+                {
+                    dprintf("USB GCR auto mode disabled\r\n");
+                    usb_gcr_auto = 0;
+                }
+                else
+                {
+                    usb_gcr_auto = 1;
+                    dprintf("USB GCR auto mode enabled\r\n");
+                }
```

Comment:
I suggest making this block of code a macro since it repeats a lot and takes up visual real estate.

```c
#define TOGGLE_BOOL(var, name) dprintf((var = !var) ? name " enabled\r\n" : name " disabled\r\n")
```

Usage: `TOGGLE_BOOL(usb_gcr_auto, "USB GCR auto mode");`
You can put the macro in a keyboard-wide or `massdrop`-wide header.

Reply:
The only exception I see to the macro would be in disabling debug mode as the message would never print since debug_enable is set to false before the dprintf. This seems to be how MAGIC_KC works though so I guess it is an expected behavior.

Reply:
Yep, the original code ([here](https://github.com/qmk/qmk_firmware/pull/3973/files#diff-fb014baff4585b4e953c103a92295c10L176), for reference) does appear to already work that way. The macro could be changed to accommodate for that:

```c
#define TOGGLE_BOOL(var, name) dprintf(!var ? name " enabled\r\n" : name " disabled\r\n"), var = !var
```

but imo the first variant is a bit cleaner. Up to you.

Reply:
Now that I think about it, this macro would look better if it were called `TOGGLE_FLAG(flag, name)`.

Reply:
The second method has the same issue but in reverse as enabling debug mode is not printed while disabling is. I propose the code just be moved to ctrl.h as follows:
```
#define TOGGLE_FLAG_AND_PRINT(var, name) { \
        if (var) { \
            dprintf(name " disabled\r\n"); \
            var = !var; \
        } else { \
            var = !var; \
            dprintf(name " enabled\r\n"); \
        } \
    }
```
I believe the _AND_PRINT is a little more descriptive for the actions involved.

Reply:
Alright, that works, and it's probably more readable too.